src/Native/Unix/System.Native/pal_signal.h

```diff
@@ -0,0 +1,59 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma once
+
+#include "pal_compiler.h"
+
+BEGIN_EXTERN_C
+
+#include "pal_types.h"
+
+/**
+ * Initializes the signal handling for use by System.Console and System.Process.
+ *
+ * Returns 1 on success; otherwise returns 0 and sets errno.
+ */
+int32_t SystemNative_InitializeSignalHandling(void);
+
+/**
+ * Hooks up the specified callback for notifications when SIGINT or SIGQUIT is received.
+ *
+ * Not thread safe.  Caller must provide its owns synchronization to ensure RegisterForCtrl
+ * is not called concurrently with itself or with UnregisterForCtrl.
+ *
+ * Should only be called when a callback is not currently registered.
+ */
+void SystemNative_RegisterForCtrl(CtrlCallback callback);
+
+/**
+ * Unregisters the previously registered ctrlCCallback.
+ *
+ * Not thread safe.  Caller must provide its owns synchronization to ensure UnregisterForCtrl
+ * is not called concurrently with itself or with RegisterForCtrl.
+ *
+ * Should only be called when a callback is currently registered. The pointer
+ * previously registered must remain valid until all ctrl handling activity
+ * has quiesced.
+ */
+void SystemNative_UnregisterForCtrl(void);
+
+typedef void (*SigChldCallback)(void);
+
+/**
+ * Hooks up the specified callback for notifications when SIGCHLD is received.
+ *
+ * Not thread safe.  Caller must provide its owns synchronization to ensure RegisterForSigChld
+ * is not called concurrently with itself.
+ *
+ * Should only be called when a callback is not currently registered.
+ */
+void SystemNative_RegisterForSigChld(SigChldCallback callback);
+
+/**
+ * Calls the original SIGCHLD handler.
+ */
+void SystemNative_ResumeSigChld(void);
```

Comment:
Nit: this name is strange to me.  What's the meaning of "resume" here?  It sounds like we paused something.

Reply:
Continue would be better, we are Continuing the handling of SIGCHLD. This method will change as part of the feedback.